backend/src/impl/default_controllers_impl.py

```diff
@@ -214,9 +214,9 @@ def systems_analyses_post(body: SystemsAnalysesBody):
         # move this function to SDK in the future or
         # add proper from_dict() methods in SDK
         for feature_name, feature in system_output_info.features.items():
-            bucket_info = feature["bucket_info"]
-            if bucket_info is not None:
-                bucket_info = BucketInfo(**bucket_info)
+            
+            bucket_info = None if feature["_type"] in set(
```

Comment:
Thanks for helping with this bug! I think this part is mostly a `from_dict()` method for `SysOutputInfo`. I'm wondering if we can move it to the SDK? It seems to be quite complicated so it may help to add unit tests for it in the future.

The `Features` class seems to be a really convenient way to allow us to specify features for each task but I think it seems a little confusing to me.
1. Could we add type annotation for `Features` class? My guess is it's something like this
```
class Features(dict[str, FeatureType])
   ...
```
If we have this, I think mypy can catch errors like this because `feature` can actually be a variety of types so we can't just assume "bucket_info" exists.
2. `FeatureType` is a union of 7 different types with very different interfaces. It also includes dict, list, and tuple which are very general types so I think that may be a little difficult to maintain. Could you please point to where these types of features are used? I'm guessing features need all kinds of attributes so we need a lot of flexibility here?
3. Instead of using `feature["_type"]`, would it achieve the same result if we use `isinstance(feature, Sequence) or isinstance(...)`? I think this may be a little safer and it also works with type checks. I am actually a little confused as why `feature["_type"]` works. I think `feature` could be a `Value` type which doesn't seem to have `__getitem__` implemented.  
4. (This is not really related to this PR. It's more of a discussion for the long term). Could we use inheritance to "group" `Sequence`, `Span`, `Value` and `Set` together so we know they have overlaps in their attributes?

Reply:
@pfliu-nlp Thanks for the PR! I agree with Lyuyang about moving it to SDK in the future. We can leave the changes here for now.

> Could we add type annotation for `Features` class? 

I agree. This is helpful

> use `isinstance(feature, Sequence)` or `instance(...)`? 

Let me rewrite the logic.

> Could we use inheritance?

I agree with this idea. it helps with serialization, too.